src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -575,19 +576,24 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                 // in the factory method call for this symbol.
                 return ((ForEachLoopBinder)this.binder).InferCollectionElementType(diagnostics, _collection);
             }
+
+            internal override SyntaxNode ForbiddenZone => _collection;
         }
 
         /// <summary>
         /// Symbol for an out variable local that might require type inference during overload resolution.
         /// </summary>
         private class PossiblyImplicitlyTypedOutVarLocalSymbol : SourceLocalSymbol
         {
+            private readonly CSharpSyntaxNode _containingInvocation;
+
             public PossiblyImplicitlyTypedOutVarLocalSymbol(
                 Symbol containingSymbol,
                 Binder binder,
                 TypeSyntax typeSyntax,
                 SyntaxToken identifierToken,
-                LocalDeclarationKind declarationKind)
+                LocalDeclarationKind declarationKind,
+                CSharpSyntaxNode containingInvocation)
```

Comment:
That (walking up) was one of the most fragile parts of the previous code for out var. The walking up is more sensitive to the syntactic context for `PossiblyImplicitlyTypedDeconstructionLocalSymbol`, but indeed I'd rather store the node in both cases. But I'd prefer to leave that for a later refactoring.

---
In reply to: [73918408](https://github.com/dotnet/roslyn/pull/12991#discussion_r73918408) [](ancestors = 73918408)